src/test/java/me/karakelley/http/handlers/StaticFilesHandlerTest.java

```diff
@@ -1,59 +1,152 @@
+
+        assertEquals(response.getStatus(), "201 Created");
+      });
+    }
+
+    @Test
+    void test409ResponseForFileThatAlreadyExists() {
+      TempFilesHelper.withTempDirectory(directory -> {
+        Path file = TempFilesHelper.createTempFile(directory, "/testing1");
+        PublicDirectory publicDirectory = PublicDirectory.create(directory.toString());
+        Handler handler = new StaticFilesHandler(publicDirectory, new HtmlPresenter());
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put("Content-Length", "9");
+        Response response = handler.respond(new Request(HttpMethod.POST, "/testing1.txt", "HTTP/1.1", headers, null, 0));
+        assertEquals("409 Conflict", response.getStatus());
+      });
+    }
   }
 
-  @Test
-  void testServesTextFile() {
-    TempFilesHelper.withTempDirectory(directory -> {
-      Path file = TempFilesHelper.createTempFile(directory, "/test1");
-      TempFilesHelper.createContents("Hello World", file);
-
-      PublicDirectory publicDirectory = PublicDirectory.create(directory.toString());
-      ContentPresenter contentPresenter = new HtmlPresenter();
-      Handler handler = new StaticFilesHandler(publicDirectory, contentPresenter);
-      Response response = handler.respond(new Request(HttpMethod.GET, "/test1.txt",  "HTTP/1.1", null, null, 0));
-      assertEquals("Hello World", new String(response.getBody()));
-    });
+  @Nested
+  class WhenRequestIsPut {
+    @Test
+    void test200ResponseForUpdatingExistingResource() {
+      TempFilesHelper.withTempDirectory(directory -> {
+        Path file = TempFilesHelper.createTempFile(directory, "/test1");
+        TempFilesHelper.createContents("Hello", file);
+        PublicDirectory publicDirectory = PublicDirectory.create(directory.toString());
+        Handler handler = new StaticFilesHandler(publicDirectory, new HtmlPresenter());
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put("Content-Length", "11");
+        Response response = handler.respond(new Request(HttpMethod.PUT, "/test1.txt", "HTTP/1.1", headers, "Hello World".getBytes(), 0));
+        assertEquals("200 OK", response.getStatus());
+      });
+    }
+
+    @Test
+    void test201ResponseForNewResource() {
+      TempFilesHelper.withTempDirectory(directory -> {
+        PublicDirectory publicDirectory = PublicDirectory.create(directory.toString());
+        Handler handler = new StaticFilesHandler(publicDirectory, new HtmlPresenter());
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put("Content-Length", "11");
+        Response response = handler.respond(new Request(HttpMethod.PUT, "/test1.txt", "HTTP/1.1", headers, "Hello World".getBytes(), 0));
+        assertEquals("201 Created", response.getStatus());
+      });
+    }
   }
-}
 
+  @Nested
+  class WhenRequestIsDelete {
+    @Test
+    void testResourceIsDeleted() {
```

Comment:
Is there a test for if the file doesn't exist?

Reply:
There is a test for this scenario in `HttpServerTest`